daliuge-engine/test/dlg_end_to_end_utils.py

```diff
@@ -65,7 +65,8 @@ def create_and_run_graph_spec(test_case: unittest.TestCase, appDropSpec: list[di
     with droputils.DROPWaiterCtx(test_case, leafs, timeout=10, expected_states=[
         DROPStates.COMPLETED, DROPStates.SKIPPED, DROPStates.ERROR, DROPStates.CANCELLED]):
         for drop in roots:
-            if app_root:
+            if drop.type != "Data":
```

Comment:
Was there any reason for modifying this? It's unclear what this is doing differently.

Reply:
This enables to have multiple mixed root nodes, which is the case in the graph I've added to the tests now.